OrbitCore/Utils.h

```diff
@@ -275,6 +275,25 @@ class CWindowsMessageToString {
 };
 #endif
 
+//-----------------------------------------------------------------------------
+inline std::string ShortenStringWithEllipsis(const std::string& text, int max_len) {
+  if (max_len < 0) {
+    return text;
+  }
+  if (max_len <= 3) {
+    return text.length() <= 3 ? text : "...";
+  }
+  if (text.length() <= max_len) {
+    return text;
+  }
+
+  const size_t chars_to_cut =
+      text.length() - static_cast<uint32_t>(max_len) + 3;
+  const size_t l = (text.length() - chars_to_cut) / 2;
+  const size_t r = l + chars_to_cut;
+  return text.substr(0, l) + "..." + text.substr(r);
```

Comment:
Thought: would the ellipsis "&mldr;" character (U+2026) be supported? You would save two characters.

Reply:
Hm maybe. But it's not so much about the actual text length but rather the size of the text on screen, so we wouldn't really benefit from this? In the end it's a bit heuristically anyways since we're not using a fixed-size font and the length in chars is just an indication and not a real measurement here.

Reply:
Yeah, I was thinking about monospace fonts, I guess that otherwise it doesn't matter.